lib/moonshot/yaml_stack_template.rb

```diff
@@ -0,0 +1,26 @@
+require 'yaml'
+require_relative 'stack_template'
+
+module Moonshot
+  # Handles YAML formatted AWS template files.
+  class YamlStackTemplate < StackTemplate
+    def initialize(filename, log:)
+      super(filename, log: log)
+      @body = YAML.load_file(@filename) if exists?
+    end
+
+    def parameters
+      @body['Parameters'].map do |k, v|
+        Parameter.new(k, v['Default'])
+      end
+    end
+
+    def resource_names
+      @body['Resources'].keys
```

Comment:
This is the same as the JSON implementation, except that it fails if there is no "Resources" key at the top level. Is that by design? If not, can we push this up to the parent class with the implementation from your JSON class?